#!/usr/bin/env python
PACKAGE = "parameter_server_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("BOOL_PARAM",   bool_t,   0, "A Boolean  parameter",  True)
gen.add("INT_PARAM",    int_t,    0, "An Integer Parameter",  1,   0, 100)
gen.add("DOUBLE_PARAM", double_t, 0, "A Double   Parameter",  0.01, 0,   1)
gen.add("STR_PARAM",    str_t,    0, "A String   parameter",  "Dynamic Reconfigure")

size_enum = gen.enum([ gen.const("Low",        int_t,  0, "Low : 0"),
                       gen.const("Medium",     int_t,  1, "Medium : 1"),
                       gen.const("High",       int_t,  2, "Hight :2")],
                       "Selection List")

gen.add("SIZE", int_t, 0, "Selection List", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "parameter_server_tutorials", "parameter_server_"))
